[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #104503
* __->__ #98826

**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

leslie-fang-intel committed:
Enable conv2d unary and binary recipe for x86 inductor quantizer

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[WIP][Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


This PR is still working in progress for internal testing.



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_quantize_pt2e.py -k TestX86InductorQuantizePT2E
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

jgong5 commented on torch/ao/quantization/_pt2e/quantizer/x86_inductor_quantizer.py:
```diff
@@ -40,6 +42,25 @@ def supported_quantized_operators() -> Dict[str, List[OperatorPatternType]]:
             [torch.nn.Conv2d],
             [F.conv2d],
+            # Conv ReLU
+            [torch.nn.Conv2d, torch.nn.ReLU],
+            [torch.nn.Conv2d, F.relu],
+            [F.conv2d, torch.nn.ReLU],
+            [F.conv2d, F.relu],
```
Is it possible to use equivalent mapping to avoid duplication here?

leslie-fang-intel replied:
I simplify it to use `itertools.product`. It can reduce much duplicated code.

jgong5 commented on torch/ao/quantization/_pt2e/quantizer/x86_inductor_quantizer.py:
```diff
@@ -154,10 +175,202 @@ def _annotate_for_static_quantization_config(
 
         return model
 
+    def _annotate_conv2d_binary_unary(
```
Possible to factor out common code from these annotation functions?

leslie-fang-intel replied:
Thanks for the suggestions. Put out common code into the help functions. And per the discussion here: https://github.com/pytorch/pytorch/pull/98730#discussion_r1229891161, we can follow up to use the common utils provided by quantization when these utils are available.

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_quantize_pt2e.py -k TestX86InductorQuantizePT2E
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_quantize_pt2e.py -k TestX86InductorQuantizePT2E
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestX86InductorQuantizePT2E
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestX86InductorQuantizePT2E
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestX86InductorQuantizePT2E
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestX86InductorQuantizePT2E
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestX86InductorQuantizePT2E
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

jerryzh168 commented on test/quantization/pt2e/test_x86inductor_quantizer.py:
```diff
@@ -68,3 +70,350 @@ def forward(self, x):
                                            expected_node_list=node_list)
+
+    @skipIfNoX86
+    def test_conv2d_unary_with_quantizer_api(self):
```
what does conv2d_unary mean? could you add a comment here?

leslie-fang-intel replied:
Thanks for the comment. Docstring added here and also for the other 5 UTs.

jerryzh168 commented on test/quantization/pt2e/test_x86inductor_quantizer.py:
```diff
@@ -68,3 +70,350 @@ def forward(self, x):
+                                               expected_node_list=node_list)
+
+    @skipIfNoX86
+    def test_conv2d_binary_with_quantizer_api(self):
```
same here

leslie-fang-intel replied:
Thanks and Docstring added for this UT.

jerryzh168 commented on test/quantization/pt2e/test_x86inductor_quantizer.py:
```diff
@@ -68,3 +70,350 @@ def forward(self, x):
+        with override_quantized_engine("x86"):
+            with torch.no_grad():
+                for inplace_relu, use_bias in itertools.product(inplace_relu_list, use_bias_list):
+                    m = Mod(inplace_relu=inplace_relu, use_bias=use_bias).eval()
+                    m_copy = copy.deepcopy(m)
+                    example_inputs = (torch.randn(2, 3, 16, 16),)
+                    # program capture
+                    m, guards = torchdynamo.export(
+                        m,
+                        *copy.deepcopy(example_inputs),
+                        aten_graph=True,
+                    )
+
+                    before_fusion_result = m(*example_inputs)
+                    import torch.ao.quantization._pt2e.quantizer.x86_inductor_quantizer as xiq
+                    quantizer = X86InductorQuantizer()
+                    operator_spec = xiq.get_default_x86_inductor_quantization_config()
+                    quantizer.set_global(operator_spec)
+                    # Insert Observer
+                    m = prepare_pt2e_quantizer(m, quantizer)
+                    after_prepare_result = m(*example_inputs)
+                    m = convert_pt2e(m)
```
wondering if you could refactor boilerplate code to a helper function, similar to https://github.com/pytorch/pytorch/blob/main/test/quantization/pt2e/test_quantize_pt2e.py#L215

leslie-fang-intel replied:
Thanks for the comments. Refactor boilerplate code as in https://github.com/pytorch/pytorch/blob/main/test/quantization/pt2e/test_quantize_pt2e.py#L215.

jerryzh168 commented on test/quantization/pt2e/test_x86inductor_quantizer.py:
```diff
@@ -68,3 +70,350 @@ def forward(self, x):
+        conv2d_type_list = [Conv2DType.left, Conv2DType.right, Conv2DType.both]
+        use_bias_list = [True, False]
+
+        with override_quantized_engine("x86"):
+            with torch.no_grad():
```
can you use two `with` in the same statement to reduce indentation?

leslie-fang-intel replied:
Thanks and changed.

jerryzh168 reviewed (approved):
looks good I think, these are the tricky cases that may not be cleaned up even when we have more utility functions support

leslie-fang-intel committed:
Update on "[Quant][PT2E] Enable conv2d unary and binary recipe for x86 inductor quantizer"


**Summary**

- Recipe to annotate `conv2d_relu` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add` for `X86InductorQuantizer` is added.
- Recipe to annotate `conv2d_add_relu` for `X86InductorQuantizer` is added.

**Test Plan**
```
python -u -m pytest -s -v test_x86inductor_quantizer.py -k TestQuantizePT2EX86Inductor
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel commented:
@pytorchbot merge
